fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,221 @@
+|---|---|---|
+|**Elements of the API, such as resources**|Boldface|**/authorization**|
+|**Variables**|Italics within curly brackets|_{ID}_|
+|**Glossary terms**|Italics on first occurrence; defined in _Glossary_|The purpose of the API is to enable interoperable financial transactions between a _Payer_ (a payer of electronic funds in a payment transaction) located in one _FSP_ (an entity that provides a digital financial service to an end user) and a _Payee_ (a recipient of electronic funds in a payment transaction) located in another FSP.|
+|**Library documents**|Italics|User information should, in general, not be used by API deployments; the security measures detailed in _API Signature_ and _API Encryption_ should be used instead.|
+
+### 1.2 Document Version Information
+
+|Version|Date|Change Description|
+|---|---|---|
+|**1.0**|2021-04-01|Initial version|
+| | | |
+
+## 2. Introduction
+
+This document details the Distributed Tracing standards to be supported for Open APIs for Interoperability (hereafter cited as the API) to ensure _tracing_ and _observability_ within the API system, and to/from API consumers.
+
+Distributed Tracing is a method to monitor a request end-to-end from the source to the destination, including any participating actors/components that process or forward the request. This helps to pinpoint where failures and poor performance of a specific request have occurred. As such this method is well suited to a modern micro-service architecture which includes distributed application components. In addition it also supports the inclusion of both API consumers which can be both internal and external (referred to as _vendors_). Vendors may participate either by continuing the trace, or forwarding the received tracing information without modification.
+
+
+## 3. Tracing Specification
+
+This section introduces Distributed Tracing from a Mojaloop context, including the header format, and the mechanisms used to generate, continue and forward traces.
+
+### 3.1 Tracing Data Model
+
+The [Table 1](#table-1) describes the trace headers fields that are optionally included in each Interoperability API request. Tables [2](#table-2), [3](#table-3), and [4](#table-4) describes the associated values and properties that is contained by each of the respective trace header fields shown in [Table 1](#table-1).
+
+**Note:** ...
+
+###### Table 1
+
+| Name | Cardinality | Type | Description | Reference |
+| --- | --- | --- | --- | --- |
+| traceparent | 1 | String(55) | Incoming request containing the trace-information, including the _version_ and populated _version-format_ values.<br/>Format: _{version}_-_{version-format}_.<br/>Example: `00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01` | Sections _[3.2.2 traceparent Header Field Values](https://www.w3.org/TR/trace-context-1/#traceparent-header-field-values)_ <sup>[1](https://www.w3.org/TR/trace-context-1/#traceparent-header-field-values)</sup> |
```

Comment:
"trace information" is used in the row below instead of "trace-information". I prefer "trace information".